core/src/test/java/apoc/convert/ConvertJsonTest.java

```diff
@@ -42,6 +44,152 @@
+                     assertJsonNode((Map<String, Object>) map.get("start"), "0", user, Map.of("name", "Adam"), null);
+                     assertJsonNode((Map<String, Object>) map.get("end"), "1", user, Map.of("name", "Jim", "age", 42L), null);
+                     assertJsonRel(map, "0", "KNOWS", Map.of("since" , 1993.1D, "bffSince", "P5M1DT12H"), RELATIONSHIP);
+        });
+    }
+
+    @Test
+    public void testToJsonPath() throws Exception {
+	    testCall(db, "CREATE p=(a:Test {foo: 7})-[:TEST]->(b:Baz {a:'b'})<-[:TEST_2 {aa:'bb'}]-(:Bar {one:'www', two:2, three: localdatetime('2020-01-01')}) RETURN apoc.convert.toJson(p) AS value",
+	             (row) -> {
+                     List<String> test = List.of("Test");
+                     List<String> bar = List.of("Bar");
+                     List<String> baz = List.of("Baz");
+                     List<Object> list = Util.fromJson((String) row.get("value"), List.class);
+
+                     assertJsonNode((Map<String, Object>) list.get(0), "0", test, Map.of("foo", 7L), NODE);
+
+                     Map<String, Object> firstRel = (Map<String, Object>) list.get(1);
+                     assertJsonNode((Map<String, Object>) firstRel.get("start"), "0", test, Map.of("foo", 7L), null);
+                     assertJsonNode((Map<String, Object>) firstRel.get("end"), "1", baz, Map.of("a", "b"), null);
+                     assertJsonRel(firstRel, "0", "TEST", null, RELATIONSHIP);
+
+                     assertJsonNode((Map<String, Object>) list.get(2), "1", baz, Map.of("a", "b"), NODE);
+
+                     Map<String, Object> secondRel = (Map<String, Object>) list.get(3);
+                     assertJsonNode((Map<String, Object>) secondRel.get("start"), "2", bar, Map.of("one", "www", "two", 2L, "three", "2020-01-01T00:00"), null);
+                     assertJsonNode((Map<String, Object>) secondRel.get("end"), "1", baz, Map.of("a", "b"), null);
+                     assertJsonRel(secondRel, "1", "TEST_2", Map.of("aa" , "bb"), RELATIONSHIP);
+
+                     assertJsonNode((Map<String, Object>) list.get(4), "2", bar, Map.of("one", "www", "two", 2L, "three", "2020-01-01T00:00"), NODE);
+	    });
+    }
+
+    @Test
+    public void testToJsonListOfPath() throws Exception {
+	    testCall(db, "CREATE p=(a:Test {foo: 7})-[:TEST]->(b:Baa:Baz {a:'b'}), q=(:Omega {alpha: 'beta'})<-[:TEST_2 {aa:'bb'}]-(:Bar {one:'www'}) RETURN apoc.convert.toJson(collect(p)+q) AS value",
+	             (row) -> {
+                     List<String> test = List.of("Test");
+                     List<String> bar = List.of("Bar");
+                     List<String> bazBaa = List.of("Baa", "Baz");
+                     List<String> omega = List.of("Omega");
+                     List<Object> list = Util.fromJson((String) row.get("value"), List.class);
+                     List<Object> firstSubList = (List<Object>) list.get(0);
+                     List<Object> secondSubList = (List<Object>) list.get(1);
```

Comment:
nit: might be worth checking there are only 2 paths, not more
(same logic applies to numbers of elements in each path)

Reply:
Added assertions